platform/error/platform-error-page-injector/src/main/java/org/codice/ddf/platform/error/injector/ErrorPageInjector.java

```diff
@@ -0,0 +1,116 @@
+
+  private Map<String, String> errorCodesMap = new HashMap<>();
+  private final String errorPagePath = "/ErrorServlet";
+
+  public ErrorPageInjector() {
+    errorCodesMap.put("400", errorPagePath);
+    errorCodesMap.put("401", errorPagePath);
+    errorCodesMap.put("403", errorPagePath);
+    errorCodesMap.put("404", errorPagePath);
+    errorCodesMap.put("405", errorPagePath);
+    errorCodesMap.put("406", errorPagePath);
+    errorCodesMap.put("500", errorPagePath);
+    errorCodesMap.put("501", errorPagePath);
+    errorCodesMap.put("502", errorPagePath);
+    errorCodesMap.put("503", errorPagePath);
+    errorCodesMap.put("504", errorPagePath);
+  }
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ErrorPageInjector.class);
+
+  @Override
+  public void event(ServiceEvent event, Map<BundleContext, Collection<ListenerInfo>> listeners) {
+    if (event.getType() == ServiceEvent.REGISTERED) {
+      Bundle refBundle = event.getServiceReference().getBundle();
+      BundleContext bundlectx = refBundle.getBundleContext();
+      Object service = bundlectx.getService(event.getServiceReference());
+      if (service instanceof ServletContext) {
+        injectErrorPage((ServletContext) service, refBundle);
+      }
+    }
+  }
+
+  private void injectErrorPage(ServletContext context, Bundle refBundle) {
+
+    Field field = null;
+    try {
+      // this grabs the enclosing instance class, which is actually a private class
+      // this is the only way to do this in Java
+      field = context.getClass().getDeclaredField("this$0");
+      field.setAccessible(true);
+    } catch (NoSuchFieldException e) {
+      LOGGER.warn(
+          "Unable to find enclosing class of ServletContext for delegating the error page. The default jetty errors will display in the browser",
+          e);
+
+      return;
+    }
+    ServletHandler handler;
+
+    // need to grab the servlet context handler so we can get down to the handler, which is what we
+    // really need
+    ServletContextHandler httpServiceContext;
+    try {
+      httpServiceContext = (ServletContextHandler) field.get(context);
+    } catch (IllegalAccessException e) {
+      LOGGER.warn(
+          "Unable to get the ServletContextHandler for {}. Jetty's default error page will be used for this context",
+          refBundle.getSymbolicName(),
+          e);
+
+      return;
+    }
+
+    // now that we have the handler, we can add in our own ErrorServlet
+    handler = httpServiceContext.getServletHandler();
+
+    ServletHolder errorServletHolder = new ServletHolder(new ErrorServlet());
+    errorServletHolder.setServletHandler(handler);
+    try {
+      errorServletHolder.start();
+      errorServletHolder.initialize();
+    } catch (Exception e) {
+      LOGGER.warn(
+          "Unable to initialize an errorServletHolder for {}", refBundle.getSymbolicName(), e);
+    }
+    handler.addServletWithMapping(errorServletHolder, errorPagePath);
```

Comment:
❓ What does it mean to add this mapping if the holder has not been started or initialized? Should we add a similar message in the warning above that Jetty's default error page will be used?

Reply:
We should return out of this method if an exception happens above because there will be major issues if the holder is not initialized.